Add support for snapd's desktop theme API

jhenstridge opened the pull request:
This PR adds support for snapd's desktop theme installation API, as found here:

https://github.com/snapcore/snapd/blob/master/daemon/api_themes.go

The API endpoint accepts GET and POST requests. The GET request is used to check the status of a list of GTK, icon, and sound themes, returning the status of each theme:
 * "installed" if an installed snap provides the theme via the content interface.
 * "available" if no installed snap provides the theme, but one exists on the store.
 * "unavailable" otherwise.

The POST request will start the install of the snaps associated with the list of theme names as an asynchronous task. It will error out if there is no work to do (i.e. all named themes are either installed or unavailable).

The GET request is open access, while the POST request requires authorisation (i.e. polkit in the case of desktop apps).

I haven't updated the Qt binding for this API yet: the PR was already quite large and I suspect there will only be a single user of the API, which will be using the glib version.

robert-ancell commented on snapd-glib/snapd-client-sync.c:
```diff
@@ -1697,3 +1697,62 @@ snapd_client_download_sync (SnapdClient *self,
+ * @cancellable: (allow-none): a #GCancellable or %NULL.
+ * @error: (allow-none): #GError location to store the error occurring, or %NULL to ignore.
+ *
+ * Check whether the status of snap packaged versions of named desktop
```
Should be "Check the status"?

robert-ancell commented on snapd-glib/snapd-client-sync.c:
```diff
@@ -1697,3 +1697,62 @@ snapd_client_download_sync (SnapdClient *self,
+ * @cancellable: (allow-none): a #GCancellable or %NULL.
+ * @error: (allow-none): #GError location to store the error occurring, or %NULL to ignore.
+ *
+ * Check whether the status of snap packaged versions of named desktop
```
Same comment as above.

robert-ancell reviewed (approved):
Apart from the small doc issue LGTM.

jhenstridge commented:
Thanks for the review.

I've updated the documentation. The body of the install_themes_sync() docs was incorrectly a copy of the check_themes_sync() docs, so I've changed that to something more appropriate.
